In [3]:
import MeCab
import unidic
print(unidic.DICDIR)
MeCab.Tagger("-d" + unidic.DICDIR)

/home/yamada/.local/lib/python3.6/site-packages/unidic/dicdir


<Swig Object of type 'MeCab::Tagger *' at 0x7f260e931e08>

- よくわからないもの

    
    

In [4]:
import json
import os
import pandas as pd
import numpy as np
from pathlib import Path
import collections
from sklearn.model_selection import train_test_split
from sklearn import metrics

import sys
sys.path.append("../")
from datatools.analyzer import *
from utterance.error_tools import *

from datatools.maneger import DataManager
from datatools.preproc import Preprocessor

In [5]:
ngram_path = "../../corpus/google/3-gram/"

In [6]:
key1_set = set( os.listdir(ngram_path) ) 

In [7]:
os.listdir(ngram_path)[:10]

['ngram_ゴストン.json',
 'ngram_オーバーニーガール.json',
 'ngram_リアテーパー.json',
 'ngram_サブスポ.json',
 'ngram_リル・フリップ.json',
 'ngram_テリーヌサラダ.json',
 'ngram_リアルタイム・リミックス.json',
 'ngram_ナゼダロウ.json',
 'ngram_ヨネン.json',
 'ngram_倥.json']

In [8]:
path = "../eval_labeled/"
datalist = ['DCM', 'DIT', 'IRS']
convs = read_conv(path, datalist)

error = "Grammatical error"
errors = ['Grammatical error', "Uninterpretable"]
sys_utt = []
y = []
for conv in convs:
    for ut in conv:
        if ut.is_system() and ut.is_exist_error():
            if not ut.utt[-1] in ["？", "！", "。", "!"]:
                sys_utt.append( clean_text( ut.utt+"。" ))
                # sys_utt.append(ut.utt)
            else:   
                sys_utt.append(ut.utt)
            if ut.is_error_included(errors):
                y.append(1)
            else:
                y.append(0)

In [9]:
print("len of y:{0}, error '{1}' counts:{2}".format(len(y), errors, y.count(1)))

len of y:1386, error '['Grammatical error', 'Uninterpretable']' counts:7


In [10]:
def load_dict(path, key1):
    filename = key_formated(key1)
    with open(path+filename, "r") as f:
        key1_dict = json.load(f) 
    return key1_dict  

In [11]:
def key_formated(key1):
    return "ngram_"+key1+".json"

In [25]:
key2dict = dict()
registerform_dict = dict()
analyzer_unidic=  Analyzer(Tokenizer())

In [60]:
#  を考慮
# 何故か↑が出てくる
def judge_key3(key1_dict, ngram):
    key1 = ngram[0]
    key2 = ngram[1]
    key3 = ngram[2]

    if key2 not in key1_dict[key1]:
        # '"'をkeyにして聞いてみる
        if '"' in key1_dict[key1] and key2 in key1_dict[key1]['"']:
            return True
        if '、' in key1_dict[key1] and key2 in key1_dict[key1]['、']:
            return True
        return False
    
    if key3 not in key1_dict[key1][key2]:
        return False

    return True 

In [61]:

def search_registerforam(key, reading):
    filename = key_formated(key)
    if filename in key1_set:
        return key
    # 登録されていない
    else:
        for other in reading:
            filename = key_formated(key)
            if filename in key1_set:
                registerform_dict[key] = other
                return other
        return key


def sentence_judge(sentence, N=3):
    
    df = analyzer_unidic.analyze_with_dataframe(sentence)
    
    # 登録
    sentence_ = ""
    for txt, read in zip(df.surface, df.reading):
        sentence_ += search_registerforam(txt, read)
    sents = sentence2docs(sentence_, sents_span=True)
    # print(sents)

    # print(tokens)
    ngram_text = []
    ngram_pos = []

    for sent in sents:
        # 各文章 : sent
        df = analyzer_unidic.analyze_with_dataframe("".join([token.orth_ for token in sent]))
        surfaces = []
        poses = []
        readings = []
        for txt, p, read in zip(df.surface, df.part_of_speech, df.reading):
            surfaces.append(txt)
            readings.append(read)
            poses.append(p)

            for i in range(len(surfaces)-N+1):
                # print(L[i:i+N])
                ngram_text.append(surfaces[i:i+N])
                ngram_pos.append(poses[i:i+N])
                # ngram_reading.append(readings[i:i+N])
    
    # print(ngram_text)
    # return
    
    for ngram, pgram in zip(ngram_text, ngram_pos):
        # if "助動詞" in pgram[1] or "助詞" in pgram[1] or "助動詞" in pgram[0] or "助詞" in pgram[0]:
        if "助動詞" in pgram[0] or "助詞" in pgram[0]:
            # print("\tcontext : {0}| ->".format(ngram))

            key1 = ngram[0]
            if key_formated(key1) not in key1_set:
                continue

            if key1 in key2dict:
                key1_dict = key2dict[key1]
            else:
                key1_dict = load_dict(ngram_path, key1)
                key2dict[key1] = key1_dict

            # チェック
            if not judge_key3(key1_dict, ngram):
                print(ngram)
                # reading を調整
                
                # 見つかれば，その語彙を登録
                return False
    return True
            


In [62]:
# df = analyzer_unidic.analyze_with_dataframe("")
for name in list(key1_set):
    if "_渾名" in name:
        print(name)

ngram_渾名.json


In [63]:
key1_dict = load_dict(ngram_path, "を")

In [64]:
for k2 in key1_dict['を']:
    if "食べる" in key1_dict["を"][k2]:
        print(k2)

"
(
)
,
-
...
<UNK>
>
[
~
“
”
、
「
」
『
【
あえて
あさって
あっさり
あっという間に
あまり
ある程度
あれこれ
あんなに
あんまり
いかに
いきなり
いくつ
いざ
いただき
いち早く
いっきに
いっぱい
いっぺんに
いつ
いつか
いつも
いやいや
いよいよ
いれ
いろいろ
うっかり
うまく
うんと
おいしく
おおいに
おそるおそる
おとなしく
おもに
おもむろに
かけ
かける
かっこよく
かなり
かりかり
がっちり
がつがつ
がばがば
がぶがぶ
がりがり
がんがん
きちっと
きちんと
きっちり
くちゃくちゃ
けっこう
こう
こうして
こっそり
こりこり
これから
こんなに
ご飯
さくっと
さして
さっさと
さっそく
さっと
さっぱり
さらに
ざくざく
し
しいて
しこたま
しっかり
しばらく
しぶしぶ
しみじみ
しめて
しょっちゅう
じっくり
じゃんじゃん
すかさず
すぐ
すこし
すごく
すばやく
すべて
する
ずっと
ずるずる
ずーっと
せっせと
ぜひ
ぜんぶ
そこそこ
そっくり
そのまま
それぞれ
それだけ
それほど
そんなに
た
たいそう
たくさん
ただ
ただただ
たっぷり
たまに
たらふく
たんと
たんまり
だいぶ
だらだら
ちびちび
ちまちま
ちゃんと
ちょうど
ちょくちょく
ちょこちょこ
ちょこっと
ちょっと
ちょっぴり
ちょびちょび
ちょびっと
ついつい
ついに
つかみ
つくり
つくる
つけ
つける
つるつる
てんこもり
てんこ盛り
で
と
とうとう
ときどき
とことん
とても
とにかく
ともに
とり
とりあえず
とる
どう
どうして
どうしても
どうにか
どっさり
どんどん
なかなか
なぜ
なるべく
なんとか
なんとなく
ぬり
の
のせ
のんびり
はじめ
はじめて
ばりばり
ぱくぱく
ぱくりと
ぱりぱり
ひたすら
ひとくち
ひとしきり
ひとつ
ひとり
ひと通り
ふたつ
ふりかけ
ふんだんに
ぺろっと
ほくほく
ほとんど
ほどほど
ぼそぼそ
ぽんぽん
まず
まずく
また
まだ
まだまだ
まったく
まったり
まるごと
まるまる
まんべんなく
まんま
みんな
むき
むさぼり
むしゃむしゃ
むしり
むりやり
めちゃくちゃ
めったに
も
もう
もう一度
もう少し
もくもく
もくもくと
もぐもぐ
もしゃもしゃ
もそもそ
も

In [65]:
for k2 in key1_dict['を']['"']:
    print(k2)

"
"(
(
-
.
0
00
1
10
100
12
15
2
20
2003
24
3
30
4
5
50
587
6
7
8
9
</S>
<UNK>
>
A
AAA
AG
Ajax
B
British
C
CR
CRU
Cal
D
E
English
Express
F
False
From
G
GP
Gain
Gedo
Global
Google
H
HI
HTML
Hanefude
I
Impact
Incomplete
J
Japan
K
L
Lord
M
MS
MSGothic
Mr
My
N
No
O
OFF
OK
ON
OR
On
P
PC
PSP
PSPmk
PSmk
Q
R
S
SE
Shift
SignUp
Studio
Superstar
T
TEXT
The
U
UMD
UTF
V
VELO
We
Web
Windows
X
Y
YES
Yahoo
Yes
Z
a
anego
ape
b
bps
brother
c
ctrl
d
dat
dokdo
e
f
false
get
great
http
i
jp
m
mbk
mile
mush
my
n
new
no
nobody
none
nowa
o
off
on
p
postmaster
r
razo
razy
read
root
s
she
su
t
tabaco
test
the
true
txt
violet
x
y
yes
»
×
φ
、
あ
あい
あいつ
あえて
あきらめ
あげる
あっ
あと
あなた
あの
あり
ありがとう
ある
ある程度
あん
い
いい
いいえ
いかに
いけ
いたずら
いち
いつ
いつも
いま
いろいろ
う
うたう
うち
うっかり
うつけ
うなる
うまく
え
お
おいしい
おいしく
おじさん
おも
および
お前
お客様
お母さん
お気に入り
お父さん
お笑い
お金
か
かく
かけ
かける
かたち
かちり
かなり
かみ殺す
から
かわいい
かわいそう
が
き
きちんと
きれい
ぎゅっ
く
くさび
くじ
くすぐる
くらす
くん
ぐるぐる
けんか
こ
こう
ここ
こだわり
こちら
こっそり
こと
こども
この
これ
これから
こんな
ご
さ
さくら
さらに
さん
し
しっかり
しめつけ
しん
じ
じゃ
す
すき
すぐ
すっきり
すべて
する
せ
そう
そこ
そして

In [66]:
"食べる" in key1_dict["を"]

False

In [67]:
y_pred_2 = []
for i, utt in enumerate( tqdm( sys_utt ) )  :
    cleaned = clean_text(utt).replace("?", "。").replace("!", "")
    # print(cleaned)
    if sentence_judge(cleaned, N=3):
        y_pred_2.append(0)
    else:
        y_pred_2.append(1)
    # if i==35:
    #     break

  1%|          | 9/1386 [00:00<00:31, 43.23it/s]

['の', '逆回り', 'は']
['と', '、', '頼り甲斐']


  2%|▏         | 25/1386 [00:00<00:28, 47.55it/s]

['の', 'ソウル', 'らしい']
['を', '受け', 'ましょう']


  4%|▍         | 53/1386 [00:01<00:29, 44.87it/s]

['を', '変え', 'ましょう']


  5%|▍         | 68/1386 [00:01<00:29, 44.40it/s]

['ね', '、', '渾名']
['ね', '、', '渾名']
['を', '渾名', 'で']
['を', '渾名', 'で']


  6%|▋         | 90/1386 [00:01<00:27, 47.44it/s]

['は', '一人', 'が']
['で', '御洒落', 'し']
['も', '頑張ろう', 'って']


  8%|▊         | 106/1386 [00:02<00:26, 48.80it/s]

['ん', 'でしょう', 'か']


  8%|▊         | 116/1386 [00:02<00:28, 45.36it/s]

['で', '御洒落', 'し']
['で', 'どんな', 'に']
['に', '格好', '付き']
['なら', 'を', '使う']


  9%|▉         | 131/1386 [00:02<00:28, 44.69it/s]

['たい', 'ん', 'でしょう']
['てる', 'ん', 'でしょう']
['を', '損ない', 'ましょう']


 12%|█▏        | 161/1386 [00:03<00:30, 40.32it/s]

['の', '朝御飯', 'メニュー']
['より', '0', '打']


 13%|█▎        | 176/1386 [00:03<00:32, 37.50it/s]

['じゃあ', '漫画', 'も']
['は', '辰年', 'です']
['です', 'か', '…']


 15%|█▍        | 201/1386 [00:04<00:28, 41.62it/s]

['と', 'ホッと', 'する']


 16%|█▌        | 218/1386 [00:04<00:24, 47.39it/s]

['は', '食', '生活']


 17%|█▋        | 233/1386 [00:05<00:24, 46.17it/s]

['の', '大きい', '福耳']


 18%|█▊        | 245/1386 [00:05<00:23, 49.34it/s]

['の', 'お', '昼']


 18%|█▊        | 255/1386 [00:05<00:24, 46.83it/s]

['で', 'テレビ', 'ある']


 19%|█▉        | 265/1386 [00:05<00:25, 44.84it/s]

['は', '一人', 'で']
['に', '欠かせ', 'て']
['は', '不', '得意']


 22%|██▏       | 311/1386 [00:06<00:24, 44.62it/s]

['た', 'を', '仕入れる']
['た', 'は', '鮮度']


 25%|██▍       | 342/1386 [00:07<00:21, 47.52it/s]

['で', 'iPhone', '0']
['へん', 'せき', 'が']


 25%|██▌       | 352/1386 [00:07<00:23, 43.95it/s]

['て', 'いき', 'ましょう']


 28%|██▊       | 382/1386 [00:08<00:24, 41.54it/s]

['で', '御洒落', 'し']


 29%|██▊       | 397/1386 [00:08<00:22, 44.02it/s]

['から', '中', '学校']
['迄', '旅行', 'に']


 29%|██▉       | 402/1386 [00:09<00:26, 37.81it/s]

['は', 'くま', 'もん']
['と', 'いう', 'アイス']
['の', 'ギョーザ', '等']
['の', 'ウッ', 'チャン']


 30%|██▉       | 414/1386 [00:09<00:31, 31.28it/s]

['た', '映画', 'フォーブス']
['は', '違和', '感']
['た', 'ドトール', 'コーヒー']
['で', 'ジャイアン', '選手']
['で', '初めて', 'エボラ']


 30%|███       | 422/1386 [00:09<00:32, 29.84it/s]

['の', '笑点', 'は']
['の', 'ファミリー', 'マート']
['に', '住む', '大渕']
['は', 'の', 'パンケーキ']
['で', 'カフェ', '坂の下']


 31%|███       | 429/1386 [00:10<00:34, 27.87it/s]

['と', 'いう', '名護']
['な', 'ゴーヤー', '等']


 31%|███       | 432/1386 [00:10<00:35, 26.86it/s]

['の', '御', '店']
['の', '御', '店']
['の', '振り掛け', '・']


 32%|███▏      | 439/1386 [00:10<00:33, 27.87it/s]

['ない', '物真似', '選手']
['の', '海遊', '館']
['た', '、', '蛸焼き']
['の', 'で', '海遊']


 32%|███▏      | 445/1386 [00:10<00:35, 26.60it/s]

['から', '、', '海遊']
['で', 'ある', 'チョコレートラテポッキーブレンド']
['の', 'チョコレートラテポッキーブレンド', 'が']
['が', '0', '数']


 33%|███▎      | 454/1386 [00:11<00:43, 21.45it/s]

['の', '子', '大粒']
['から', 'でしょう', 'か']
['は', 'コロ', 'プラ']
['の', 'shall', 'ウィー']
['の', 'で', 'ミスター']
['の', 'は', '一握り']


 33%|███▎      | 457/1386 [00:11<00:42, 21.79it/s]

['で', 'の', '饂飩']
['の', 'キャンデー', 'Microsoftband']
['の', 'キャンデー', 'Microsoftband']
['で', 'ある', 'Microsoftband']


 33%|███▎      | 463/1386 [00:11<00:42, 21.97it/s]

['の', '里田', 'まい']
['で', 'ある', 'Microsoftband']
['だ', 'から', '荒野']
['タ', 'モリ', 'さん']
['の', '町歩き', 'テレビ']


 34%|███▍      | 470/1386 [00:11<00:35, 25.49it/s]

['の', '町歩き', 'テレビ']
['は', '、', 'ashtsujiguchi']
['な', 'スイーツ', '・']
['れる', 'ミディアム', '霊能']


 35%|███▍      | 480/1386 [00:12<00:33, 26.78it/s]

['た', 'お', '節']
['が', '、', '糠床']
['は', 'は', 'ハードル']
['の', '御', '馳走']
['の', '町歩き', 'テレビ']
['タ', 'レント', 'の']


 35%|███▍      | 483/1386 [00:12<00:38, 23.28it/s]

['と', 'いう', 'ホール']
['た', 'ワールド', 'カップ']
['た', 'イングレス', 'の']


 35%|███▌      | 490/1386 [00:12<00:34, 25.98it/s]

['から', 'はずれ', '淋しく']
['の', '世界', 'ナゼ']
['て', 'も', '仕方ない']
['が', '楽しい', 'でしょう']


 36%|███▌      | 499/1386 [00:13<00:35, 24.99it/s]

['タ', 'モリ', 'さん']
['の', '町歩き', 'テレビ']
['の', '八木', '竜']


 36%|███▋      | 505/1386 [00:13<00:43, 20.40it/s]

['の', '中華', 'まん']
['の', 'ハイ', 'チュウ']
['は', 'ヨーク', 'ベニマル']
['は', '表', '参道']
['に', '表', '参道']


 37%|███▋      | 508/1386 [00:13<00:40, 21.75it/s]

['の', '村上', '信五']
['の', '村上', '信五']
['が', '持つ', 'レギュラー']
['は', '、', '牛タン']
['は', '所々', 'たくさん']


 37%|███▋      | 514/1386 [00:13<00:37, 23.56it/s]

['と', '褌姿', 'は']
['と', 'いう', 'ドラマ']
['が', '、', '糠床']
['だ', 'から', '荒野']


 38%|███▊      | 520/1386 [00:14<00:34, 25.45it/s]

['と', 'いう', '旅']
['タ', 'レント', 'の']
['の', 'ブラタモリ', 'で']
['の', '閻', '涵']


 38%|███▊      | 527/1386 [00:14<00:31, 27.35it/s]

['の', 'や', '別ち']
['たい', '都市', '0']
['か', 'も', 'ん']


 38%|███▊      | 533/1386 [00:14<00:31, 26.72it/s]

['か', 'も', 'ん']
['が', '、', '日清']
['の', 'ソレデモシタイ', 'の']
['で', 'は', '日清']


 39%|███▉      | 540/1386 [00:14<00:29, 28.81it/s]

['の', '子', '大粒']
['は', '結構', '太麺']
['も', 'サッカー', 'オーストラリア']
['が', 'シドニー', '球界']
['の', 'シドニー', '球界']


 39%|███▉      | 544/1386 [00:14<00:28, 29.95it/s]

['が', 'シドニー', '代表']
['は', 'サッポロ', 'ビール']


 40%|████      | 557/1386 [00:15<00:32, 25.63it/s]

['も', 'サッカー', 'オランダ']
['は', 'ジャニーズ', 'の']
['たい', '有名', '人']
['て', 'いる', 'ジャニーズ']
['の', '忘年', '会']


 41%|████      | 563/1386 [00:15<00:31, 26.15it/s]

['と', 'いう', 'スイーツ']
['の', '御薩', 'スナック']
['の', 'ちょい', '足し']
['に', '印刷', '物']
['と', 'いう', 'スイーツ']


 41%|████      | 569/1386 [00:15<00:30, 26.55it/s]

['の', '0', '週間']
['を', '0', '名']
['が', '和洋', '折衷']
['の', 'かっぱ', '蝦煎']


 42%|████▏     | 579/1386 [00:16<00:27, 28.94it/s]

['が', '楽しい', 'でしょう']
['よう', 'な', 'かば']
['の', 'グランド', 'カルビー']
['の', '力め', 'し']
['と', 'いう', '円柱']


 42%|████▏     | 585/1386 [00:16<00:30, 26.49it/s]

['が', '桜ん坊', 'の']
['の', '0', '県']
['の', '0', '県']
['に', '、', 'こて']


 43%|████▎     | 592/1386 [00:16<00:28, 28.33it/s]

['の', 'で', '滅多']
['の', 'ハイ', 'チュウ']
['の', '広島', '鉄']
['たい', 'の', 'でしょう']
['は', '美味しい', 'ナゴヤ']
['の', '安納', '薯']


 43%|████▎     | 598/1386 [00:17<00:37, 20.99it/s]

['の', '安納', '薯']
['の', 'ファミマ', 'プレミアム']
['の', '、', 'ウォーズ']
['は', 'メイ', 'トー']
['の', '羽生', '結']


 43%|████▎     | 601/1386 [00:17<00:36, 21.50it/s]

['も', 'サッカー', 'オーストラリア']
['で', 'アカデミー', '名誉']
['の', 'サッカー', 'ホンジュラス']
['の', '田中', '将大']


 44%|████▍     | 610/1386 [00:17<00:31, 24.89it/s]

['な', 'サンタ', 'クロース']
['の', 'ミスター', 'クロワッサン']
['の', 'ミスター', 'クロワッサン']
['て', 'くれる', 'でしょう']
['の', '子', '大粒']


 44%|████▍     | 616/1386 [00:17<00:28, 26.94it/s]

['れる', 'ウルトラ', 'ヒーロー']
['の', '馬事', '公園']


 45%|████▍     | 622/1386 [00:18<00:31, 24.15it/s]

['の', 'で', '凄い']
['と', 'エルサ', 'の']
['に', '日本', 'エレキテル']
['で', 'ある', 'モンスターズ']


 45%|████▌     | 628/1386 [00:18<00:33, 22.51it/s]

['の', 'ハビエル', '・']
['と', '0', 'FIFA']
['から', 'はずれ', '淋しく']
['の', 'サッカー', '横浜']
['も', '0', '位']


 46%|████▌     | 632/1386 [00:18<00:29, 25.44it/s]

['の', 'じゃがりこ', '味']
['な', '御', '店']
['の', 'グランド', 'カルビー']
['て', 'ハロー', 'キティー']


 46%|████▌     | 635/1386 [00:18<00:30, 25.02it/s]

['と', 'いう', 'テレビ']
['と', 'いう', 'テレビ']


 46%|████▌     | 641/1386 [00:18<00:38, 19.13it/s]

['で', '年末', '年始']
['の', 'ザ', 'manzai']
['と', 'いう', 'ドラマ']
['た', '0', 'g']
['の', '痛快', '明石家']


 46%|████▋     | 644/1386 [00:19<00:36, 20.18it/s]

['と', 'ナインティー', 'ナイン']
['たら', '本当', 'は']
['から', 'でしょう', 'か']
['や', '田丸', '屋']


 47%|████▋     | 654/1386 [00:19<00:31, 22.99it/s]

['の', 'あまおう', 'の']
['な', '0', '組']
['れる', 'カバーアルバムキャンニュキープ', 'a']
['な', '0', '組']
['など', 'が', 'You']


 47%|████▋     | 657/1386 [00:19<00:30, 23.75it/s]

['は', '現', '時点']
['て', 'シンガポール', 'プロ']
['が', 'シンガポール', '球界']
['と', 'いう', 'ホテル']
['は', 'とくに', '松坂']


 48%|████▊     | 664/1386 [00:19<00:29, 24.85it/s]

['の', 'カミング', 'アウト']
['の', 'カミング', 'アウト']
['の', '御', '店']
['の', 'カミング', 'アウト']
['の', '漆塗り', 'が']


 49%|████▊     | 673/1386 [00:20<00:27, 25.52it/s]

['は', 'もちろん', '付け麺']
['で', 'は', '鸚哥']
['の', '水族', '館']
['て', 'から', 'ディフェンダー']
['と', 'いう', 'ドイツ']


 49%|████▉     | 680/1386 [00:20<00:25, 27.82it/s]

['の', 'EU', '戦']
['の', 'サッカー', 'コロンビア']
['た', 'かば', '館']


 50%|████▉     | 687/1386 [00:20<00:24, 28.62it/s]

['の', '御', '店']
['よう', 'な', 'かば']
['の', '妖怪', 'ウォッチ']
['の', 'フェルメール', 'さん']


 50%|█████     | 693/1386 [00:20<00:26, 26.37it/s]

['や', 'ハビエル・アギーレ', 'など']
['の', 'ココ', 'ロココ']
['の', 'ココ', 'ロココ']
['の', '世界', 'ナゼ']


 50%|█████     | 699/1386 [00:21<00:26, 25.96it/s]

['た', 'の', 'プラモデル']
['の', 'BANDAI', 'ホビー']
['の', 'BANDAI', 'ホビー']
['と', 'いう', '炭酸']


 51%|█████     | 708/1386 [00:21<00:25, 26.14it/s]

['の', 'ドーナツ', '0']
['の', '本田', '圭佑']
['の', '本田', '圭佑']
['が', '三都主', 'アレッサンドロ']


 52%|█████▏    | 714/1386 [00:21<00:33, 20.20it/s]

['の', 'リプトン', '・']
['で', '銀', 'メダル']
['まで', 'FIFA', 'ワールド']
['か', '鈴木', '明子']


 52%|█████▏    | 721/1386 [00:22<00:27, 24.01it/s]

['で', 'ある', 'イングレス']
['より', 'もう', '少し']
['に', '島歌', 'と']
['で', '、', 'グラッチェ']
['は', '映画', '煎']


 52%|█████▏    | 727/1386 [00:22<00:28, 23.31it/s]

['は', '、', '数十']
['ながら', 'みけ', 'らん']
['に', '御嶽', '山']


 53%|█████▎    | 733/1386 [00:22<00:26, 24.45it/s]

['て', 'フジ', 'テレビ']
['で', '0', '月']
['と', 'いう', 'ドラマ']
['ない', '物真似', '選手']
['は', 'なかなか', '漫才']
['か', 'くま', 'もん']
['の', 'リプトン', '・']


 53%|█████▎    | 739/1386 [00:22<00:26, 24.28it/s]

['の', 'ミッドフィルダー', '本田']
['の', 'セレッソ', '大阪']
['に', '遠藤', '保仁']


 54%|█████▍    | 746/1386 [00:23<00:25, 25.58it/s]

['も', '神', '様']
['に', 'アップロード', 'し']
['の', 'ヒカ', 'キン']


 55%|█████▍    | 756/1386 [00:23<00:21, 29.58it/s]

['の', 'が', 'ツイッター']
['の', 'エイプリル', 'フール']
['が', '楽しい', 'でしょう']
['の', 'サザン', 'オール']


 55%|█████▍    | 762/1386 [00:23<00:23, 27.12it/s]

['など', '茅ヶ崎', '地方']
['は', 'ワールド', 'カップ']
['も', 'ブラタモリ', '見']
['の', '町歩き', 'テレビ']
['か', '森田', '芳光']


 55%|█████▌    | 769/1386 [00:23<00:21, 28.84it/s]

['て', '致命', '傷']
['は', '海外', '力士']
['が', 'いい', 'でしょう']
['に', 'よる', '0']


 56%|█████▌    | 776/1386 [00:24<00:22, 27.18it/s]

['なら', '燐', '・']
['も', '燐', '・']


 56%|█████▋    | 782/1386 [00:24<00:23, 25.46it/s]

['の', '宮本', '恒靖']
['の', '宮本', '恒靖']
['の', '川澄', '奈穂美']
['の', 'ゴール', 'キーパー']


 57%|█████▋    | 788/1386 [00:24<00:23, 25.51it/s]

['の', 'アンデルソン', '・']
['は', 'ガンバ', '大阪']
['や', '高級', '焼き肉']
['が', '総額', '000000000']


 57%|█████▋    | 794/1386 [00:24<00:23, 25.26it/s]

['の', 'ソレデモシタイ', 'は']
['の', 'バラード', 'SHOW']
['の', 'フィリピン', 'カレー']
['と', 'いう', '音楽']


 58%|█████▊    | 800/1386 [00:25<00:30, 19.50it/s]

['と', 'いう', 'サッカー']
['の', '侭', 'は']
['は', 'テレビ', '朝日']
['は', '、', 'ナインティー']
['れる', 'レコ', 'チョク']
['の', 'ぶー', '社']


 58%|█████▊    | 806/1386 [00:25<00:25, 22.87it/s]

['の', 'ぶー', '社']
['が', '桜ん坊', 'の']
['くらい', 'フーズ', 'に']
['くらい', 'フーズ', 'の']


 59%|█████▊    | 813/1386 [00:25<00:22, 25.52it/s]

['の', 'ソントン', 'ホールディングス']
['は', 'アイドル', 'グループス']
['も', 'とても', 'ヒット']
['な', '0', '組']


 59%|█████▉    | 816/1386 [00:25<00:23, 24.63it/s]

['な', '0', '組']
['と', 'いう', 'CD']
['に', 'よる', '0']
['と', 'いう', 'CD']


 59%|█████▉    | 822/1386 [00:26<00:24, 23.05it/s]

['で', '国営', '滝野']
['から', '夜更かし', 'で']
['を', 'マツコ', 'の']
['と', 'いう', '技']


 60%|█████▉    | 828/1386 [00:26<00:23, 23.35it/s]

['の', '鸚哥', '味']
['の', 'グランド', 'カルビー']
['の', 'ぐり', 'こ']
['が', '少な', '目']
['の', 'ジャイアントカプリコ', '苺味']


 60%|██████    | 836/1386 [00:26<00:21, 26.15it/s]

['の', 'ジャイアントカプリコ', '苺味']
['の', '広島', '鉄']
['なら', '、', '生チョコ']
['に', 'ポテト', 'チップス']


 61%|██████    | 842/1386 [00:27<00:20, 26.05it/s]

['の', 'スナック', '菓子']
['で', 'アサヒ', 'ビール']
['の', 'スナック', '菓子']
['の', 'かっぱ', '蝦煎']
['の', '三都主', 'アレッサンドロ']
['た', '国際', 'アンデルセン']


 61%|██████    | 848/1386 [00:27<00:21, 24.59it/s]

['と', 'いう', 'ポップコーン']


 62%|██████▏   | 854/1386 [00:27<00:20, 25.79it/s]

['や', '香川', '真司']
['て', 'カミング', 'アウト']
['も', '世界', '選手']
['の', '漆塗り', 'が']


 62%|██████▏   | 860/1386 [00:27<00:19, 26.91it/s]

['の', 'ファミリー', 'マート']
['な', 'コンビニエンス', 'ストア']
['の', 'fukase', 'さん']
['は', 'まだ', 'エボラ']


 63%|██████▎   | 868/1386 [00:27<00:16, 30.54it/s]

['に', '0', '週間']
['ば', '、', 'letao']
['で', '下', '半身']


 63%|██████▎   | 876/1386 [00:28<00:22, 22.98it/s]

['の', '0', '種類']
['が', '桜ん坊', 'の']


 64%|██████▎   | 883/1386 [00:28<00:19, 26.09it/s]

['し', '、', 'グラッチェ']
['だ', 'から', '荒野']


 64%|██████▍   | 890/1386 [00:28<00:18, 27.26it/s]

['が', '鈴木', '章']
['の', 'もやもや', 'さまぁ']
['に', '、', 'サザンアイルズ']


 65%|██████▍   | 896/1386 [00:29<00:18, 26.05it/s]

['から', '、', 'サザンアイルズ']
['乃', '逢', 'の']
['は', '特に', 'pasco']
['が', 'pasco', 'の']
['で', 'は', 'pasco']


 65%|██████▌   | 903/1386 [00:29<00:17, 27.99it/s]

['の', '信長', '協奏']
['まで', 'なら', 'アナウンサー']


 66%|██████▌   | 911/1386 [00:29<00:16, 29.39it/s]

['の', '鸚哥', '味']
['の', '鸚哥', '味']
['の', '鸚哥', '味']
['の', '鸚哥', '味']


 66%|██████▋   | 921/1386 [00:30<00:15, 29.09it/s]

['の', '生姜', 'チャイ']
['の', '生姜', 'チャイ']
['て', 'いる', '町作り']
['で', 'モ', 'スライス']
['も', 'MOS', 'バーガー']


 67%|██████▋   | 927/1386 [00:30<00:17, 26.63it/s]

['と', 'エルサ', 'の']
['で', '恋', 'する']
['の', 'akb', '0']
['の', '軍師', '官兵衛']
['の', '軍師', '官兵衛']


 67%|██████▋   | 933/1386 [00:30<00:18, 24.89it/s]

['で', '、', 'akb']
['な', '岡崎', '慎司']
['で', 'くつろぐ', '画像']
['が', '岡崎', '慎司']


 68%|██████▊   | 941/1386 [00:30<00:14, 30.87it/s]

['の', 'で', '厄介']


 68%|██████▊   | 949/1386 [00:31<00:16, 25.86it/s]

['で', 'ぎりぎり', '何']
['に', 'ある', '御洒落']
['て', '、', '香り付け']
['と', 'か', 'スープ']


 70%|██████▉   | 966/1386 [00:31<00:13, 30.31it/s]

['か', 'も', '必須']
['です', 'ね', '一人']
['と', 'か', '天国']
['か', '.', '牛タン']


 70%|███████   | 974/1386 [00:31<00:12, 33.04it/s]

['や', '、', '学祭']
['ね', 'ー', 'ソフト']
['ながら', 'お', 'しまい']


 71%|███████   | 987/1386 [00:32<00:11, 36.12it/s]

['を', 'し', 'ましょう']
['と', 'か', 'いる']
['に', '読む', '積もり']


 72%|███████▏  | 1000/1386 [00:32<00:09, 38.83it/s]

['と', 'か', '操作']
['か', '…', 'どじ']
['ず', 'タイピング', 'し']


 74%|███████▎  | 1019/1386 [00:33<00:09, 39.21it/s]

['か', '…', 'どじ']


 74%|███████▍  | 1028/1386 [00:33<00:09, 36.28it/s]

['ます', 'そう', 'いう']
['てる', 'ん', 'でしょう']
['を', '遣い', 'ながら']
['です', 'か', '大晦日']


 75%|███████▌  | 1040/1386 [00:33<00:10, 33.03it/s]

['の', '0', '週間']
['も', '0', '週間']
['す', '…', '凄い']


 76%|███████▌  | 1053/1386 [00:33<00:08, 37.11it/s]

['は', 'やみつき', 'に']
['た', 'くらい', 'でしょう']


 77%|███████▋  | 1065/1386 [00:34<00:10, 31.28it/s]

['た', '「', 'あずみ']
['です', 'かー', '。']


 78%|███████▊  | 1079/1386 [00:34<00:08, 35.94it/s]

['の', 'ジャズ', 'セクション']
['と', 'か', '自然']
['の', 'クライン', 'キー']
['に', 'なろう', 'と']
['と', 'か', '自然']


 79%|███████▉  | 1092/1386 [00:35<00:08, 34.14it/s]

['って', '陰', 'から']
['でし', 'た', '…']
['か', 'も', '必須']
['か', 'が', 'ひどかっ']


 79%|███████▉  | 1101/1386 [00:35<00:07, 36.63it/s]

['は', '酎ハイ', '程度']
['は', '酎ハイ', '程度']
['じゃあ', 'あまり', 'お']
['よ', 'ね', '…']


 80%|████████  | 1114/1386 [00:35<00:07, 37.32it/s]

['と', 'か', '表情']
['が', '切磋', '琢磨']


 81%|████████  | 1118/1386 [00:35<00:09, 29.64it/s]

['に', '読む', '積もり']
['の', '踏み場', 'は']


 82%|████████▏ | 1131/1386 [00:36<00:07, 36.23it/s]

['が', 'やたら', '粘る']
['で', '忘年', '会']
['です', '.', '。']
['と', 'いう', '斬新']


 82%|████████▏ | 1143/1386 [00:36<00:06, 36.24it/s]

['が', '声高い', '子']
['に', '瀕し', 'て']
['てる', 'ん', 'でしょう']


 83%|████████▎ | 1153/1386 [00:36<00:05, 40.06it/s]

['な', 'ん', 'でしょう']
['です', 'ね', '…']


 84%|████████▎ | 1158/1386 [00:36<00:05, 38.24it/s]

['と', 'か', '天国']


 84%|████████▍ | 1171/1386 [00:37<00:06, 33.64it/s]

['が', '若槻', 'っぽい']
['ない', 'ん', 'でしょう']


 85%|████████▌ | 1180/1386 [00:37<00:05, 36.21it/s]

['て', '焦がし', 'たい']
['の', '白土', '人']
['から', 'そう', '急かす']


 87%|████████▋ | 1199/1386 [00:38<00:04, 39.86it/s]

['たら', '猫好き', 'に']
['の', '風物', '詩']
['だ', 'か', '取り留め']


 87%|████████▋ | 1209/1386 [00:38<00:04, 38.06it/s]

['で', '済ま', 'せよう']
['っけ', '、', '時代']
['です', 'けど', '名店']


 88%|████████▊ | 1224/1386 [00:38<00:03, 41.38it/s]

['なんか', 'ダイソン', 'です']


 89%|████████▉ | 1234/1386 [00:38<00:03, 39.49it/s]

['は', '痛み', 'にく']
['で', 'たまに', 'マージャン']


 90%|████████▉ | 1243/1386 [00:39<00:04, 35.53it/s]

['れる', 'こと', 'でしょう']
['よっ', '笑い', '。']


 90%|█████████ | 1252/1386 [00:39<00:03, 36.21it/s]

['も', 'オリンピック', '書い']
['す', 'っ', 'っ']
['に', '一人', 'だけ']
['と', '一人', 'で']


 91%|█████████ | 1260/1386 [00:39<00:03, 34.06it/s]

['と', 'か', '外']
['て', '食べれ', 'ます']


 91%|█████████▏| 1268/1386 [00:40<00:03, 32.99it/s]

['の', 'で', '短所']
['な', 'マスコット', 'キャラクター']


 93%|█████████▎| 1285/1386 [00:40<00:02, 34.52it/s]

['を', '0', 'くらい']
['も', 'さっさと', '美味しい']
['に', 'は', '敵い']
['です', 'かー', '。']
['です', 'か', '年始']


 93%|█████████▎| 1293/1386 [00:40<00:02, 35.73it/s]

['から', '段', 'ボール']
['ほど', 'あり', '一気']
['か', 'も', '必須']


 94%|█████████▍| 1301/1386 [00:40<00:02, 32.69it/s]

['と', 'か', '日本']
['と', 'か', '絶対']


 94%|█████████▍| 1309/1386 [00:41<00:02, 33.65it/s]

['そう', 'す', 'ね']
['へ', '通っ', 'てらっしゃる']
['と', 'いう', 'オーストラリア']
['が', '切磋', '琢磨']
['ほど', '0', '年']


 95%|█████████▌| 1318/1386 [00:41<00:01, 34.36it/s]

['な', 'ん', 'でしょう']
['が', '切磋', '琢磨']
['みたい', 'に', 'かなう']
['た', 'から', '…']


 96%|█████████▌| 1327/1386 [00:41<00:01, 35.76it/s]

['の', 'で', 'ガッツリ']
['と', '黒猫', 'です']


 96%|█████████▋| 1336/1386 [00:41<00:01, 37.93it/s]

['も', 'ここ', '0']
['は', '大変', 'でしょう']


 97%|█████████▋| 1344/1386 [00:42<00:01, 36.64it/s]

['みたい', 'に', 'かなう']
['で', '…', '。']
['か', 'は', '作り置き']


 98%|█████████▊| 1361/1386 [00:42<00:00, 36.33it/s]

['なら', '"', '良い']
['でし', 'た', '…']
['た', 'くらい', 'でしょう']
['が', '勉強', '学校']


 99%|█████████▉| 1369/1386 [00:42<00:00, 36.72it/s]

['なんて', '個人', 'も']
['と', 'か', '暇']
['の', '踏み場', 'は']


 99%|█████████▉| 1377/1386 [00:43<00:00, 37.46it/s]

['の', '踏み場', 'は']
['は', '仕方ない', 'です']


100%|██████████| 1386/1386 [00:43<00:00, 31.99it/s]

['た', '飛行', '機']


In [68]:
y_pred_2.count(0)

893

In [69]:
score(y, y_pred_2)

confusion matrix = 
 [[887 492]
 [  6   1]]
accuracy =  0.6406926406926406
precision =  0.002028397565922921
recall =  0.14285714285714285
f1 score =  0.003999999999999999


- 使い方が難しい

        confusion matrix = 
        [[666 713]
        [  6   1]]
        accuracy =  0.48124098124098125
        precision =  0.0014005602240896359
        recall =  0.14285714285714285
        f1 score =  0.0027739251040221915

- ちょっと改善(無理やり)

        confusion matrix = 
        [[887 492]
        [  6   1]]
        accuracy =  0.6406926406926406
        precision =  0.002028397565922921
        recall =  0.14285714285714285
        f1 score =  0.003999999999999999

    - 意外と普通っぽい ngram がない
        - ["を", "覚える", *]

    - スコア化はせず， ngram が存在しなければアウトと仮定した
        - 先行研究のモデルで，そういったボーダーは存在しない

In [70]:
mecab_tokenize("お兄さんが死んだ")

['お', '兄さん', 'が', '死ん', 'だ']

In [71]:
df = analyzer_unidic.analyze_with_dataframe("頼りがいのあった可愛くない兄さんが死んだ")
df

surface part_of_speech infl_type infl_form base_form  \
0     頼りがい    名詞-普通名詞-一般-                         タヨリガイ   
1        の       助詞-格助詞--                             ノ   
2       あっ     動詞-非自立可能--     五段-ラ行   連用形-促音便        アル   
3        た         助動詞---     助動詞-タ    連体形-一般         タ   
4      可愛く       形容詞-一般--       形容詞    連用形-一般      カワイイ   
5       ない    形容詞-非自立可能--       形容詞    連体形-一般        ナイ   
6        兄    名詞-普通名詞-一般-                            アニ   
7       さん    接尾辞-名詞的-一般-                            サン   
8        が       助詞-格助詞--                             ガ   
9       死ん        動詞-一般--     五段-ナ行   連用形-撥音便        シヌ   
10       だ         助動詞---     助動詞-タ    連体形-一般         タ   

                             reading phonetic  
0   [頼り甲斐, 頼りがい, タヨリガイ, 頼りがい, タヨリガイ]        *  
1                    [の, の, ノ, の, ノ]        *  
2               [有る, あっ, アッ, ある, アル]        *  
3                    [た, た, タ, た, タ]        *  
4        [可愛い, 可愛く, カワイク, 可愛い, カワイー]        *  
5               [無い, ない, ナイ, ない, ナイ]        *  
6                  [兄, 兄, アニ, 兄, アニ]        *  
7               [さん, さん, サン, さん, サン]        *  
8                    [が, が, ガ, が, ガ]        *  
9               [死ぬ, 死ん, シン, 死ぬ, シヌ]        *  
10                   [た, だ, ダ, だ, ダ]        *